### ISSUE TITLE
[aws-cloudtrail] cross-account `Trail`s cannot specify SNS topic

### ISSUE THREAD

**Issue by NGL321** [aws/aws-cdk#11387]
_Continuation of: #9013_
_References internal ticket_

When creating a new trail using an SNS topic from secondary account (see code sample below). Trail creation fails.

``` ts
const externAccountTopic = sns.Topic.fromTopicArn(
   this, 
   'TopicName',
   'TOPIC_ARN_SECONDARY_ACCOUNT'
);

const multiAccountTrail = new cloudtrail.Trail(this, 'MultiAccountTrail', {
   bucket: dataEventTrailBucket,
   enableFileValidation: true,
   isMultiRegionTrail: true,
   snsTopic: externAccountTopic,
   trailName: 'MultiAccountTrail'
});

```

Fails with exception:
```
"SNS Topic does not exist or the topic policy is incorrect! (Service: AWSCloudTrail; Status Code: 400; Error Code: InsufficientSnsTopicPolicyException; Request ID: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx; Proxy: null)"
```

### Related Issue

It looks like this was brought-up before, and largely addressed (initial concern was multi-region), but not fixed for multi-account.

Original Issue: #9013 
Fixing PR: #9239

### Environment

  - **CDK CLI Version  :** 1.72.0
  - **Framework Version:** 1.72.0
  - **Node.js Version:**  12.16.2
  - **OS               :** OSX Catalina
  - **Language (Version):**  Typescript

### Other

--- 

This is :bug: Bug Report

**Comment by tguedes-lastwall** [comment]
This is still an issue in 2.92.0

**Comment by FilipPyrek** [comment]
Yeah, this issue is still there. `Trail` construct still doesn't work for multi account-multi region setup.

**Comment by mrgrain** [comment]
Thanks for reporting. Re-opened.

**Comment by peterwoodworth** [comment]
Yes, if you take the example in the initial post I'm getting a separate error after using an escape hatch to specify an arn. Here's the code,

```ts
const externAccountTopic = sns.Topic.fromTopicArn(
   this, 
   'TopicName',
   'TOPIC_ARN_SECONDARY_ACCOUNT'
);

const multiAccountTrail = new cloudtrail.Trail(this, 'MultiAccountTrail', {
   bucket: dataEventTrailBucket,
   enableFileValidation: true,
   isMultiRegionTrail: true,
   snsTopic: externAccountTopic,
   trailName: 'MultiAccountTrail'
});

(multiAccountTrail.node.defaultChild as cloudtrail.CfnTrail).addPropertyOverride('SnsTopicName', 'TOPIC_ARN_SECONDARY_ACCOUNT');
```

And here's the error:
<img width="1236" alt="Screenshot 2023-09-13 at 4 15 09 PM" src="https://github.com/aws/aws-cdk/assets/44349620/c914264b-f3ee-435a-9b0d-d63ff4bb3867">

It's a separate error message than if the topic arn didn't exist, however I'm unable to figure out why this is throwing here.

Since the `Topic` is in a separate account, the call to create the topic policy that permits CloudTrail to publish will not succeed. This will have to be added to your other account if it is not already created. I thought this was what was causing the error, but manually adding it to the `Topic` in the other account didn't get me past the error. I'm likely missing some other policy that I need, will be able to check tomorrow

**Comment by FilipPyrek** [comment]
Maybe try classic SNS topic, not FIFO @peterwoodworth. Classic SNS topic worked for me.

**Comment by peterwoodworth** [comment]
Yep, seems to work on classic for me! In that case the code block above works as a workaround, and we cannot create the policy for this on the Topic which has to be done ahead of time, but we can pass the arn to CloudFormation instead of just the name to make this work without an escape hatch.

**Comment by HCrane** [comment]
Just wanting to push as i stumbled upon this issue as of right now! Keep up the good work!

**Comment by RichardMills** [comment]
We ran into this when setting up a topic in account1 and using `topicFromArn()` in account2 to create the Trail similar to the code shown in this issue.

I think the core problem is here:

https://github.com/aws/aws-cdk/blob/0c98552723a912f7e05548b70f4ae3e64f6fa6e2/packages/aws-cdk-lib/aws-cloudtrail/lib/cloudtrail.ts#L353

Where it always uses `topicName` and thus won't include the account number and other components of the ARN.
```
      snsTopicName: this.topic?.topicName,
```

This should use `this.topic.topicArn` instead of `topicName` to provide a portable reference that works regardless of local account or cross-account. 

I ended up scrapping the use of `cloudtrail.Trail()` and just implemented mine using the L1 `CfnTrail()` low level construct. 

Minor update: I did not test whether the `addPropertyOverride()` trick would work in my case because I found this issue after I implemented my code.

I was not able to determine why using a "Classic" vs. "FIFO" topic would make this any different.
